### ISSUE TITLE
Error 503 trying to get candles from Bittrex

### ISSUE THREAD

**Issue by RikardoPons** [dparlevliet/node.bittrex.api#61]
I tried to get the candles from bittrex but I got Error 503 always...
I opened the website of bittrex and the website ask me for captcha verification. 
It is the same in all the request I do.

  Thanks!

**Comment by samuelcardillo** [comment]
I am getting the same constantly and sometimes it works. I guess Bittrex is having some issue.

**Comment by dparlevliet** [comment]
Yeah it sounds like we need to put cloudscraper on the normal calls, too - but unfortunately I'm not getting this issue so it'd be hard for me to implement it and know for sure that the issue is solved. Do either of you think you could have a bash at it?

**Comment by samuelcardillo** [comment]
When you said cloudscraper @dparlevliet I looked at the code of the module I had and that was the problem... It was the deprecated one and not the one from your fork. Removed it and `npm i` yours instead: working again.

**Comment by skyl** [comment]
I get a 503 when I try to connect to the websocket feed: https://github.com/ericsomdahl/python-bittrex/issues/57#issuecomment-343769903

Is anyone else experiencing this?

Did Bittrex just add a connection token to the request? I didn't notice it before:

```
wss://socket.bittrex.com/signalr/connect?transport=webSockets&clientProtocol=1.5&connectionToken=28%2BotNt6hw6w%2BaNTgvve4ME4aTHBf5Mk2niqkjohrE3yNic3yMX88N8q%2F3DYcoOlNdE4W8tuqy1vR8ssLLaMQ0YKfUkCJ%2B0UsiI%2B8A6b&connectionData=%5B%7B%22name%22%3A%22corehub%22%7D%5D&tid=1
```

**Comment by dparlevliet** [comment]
They have enabled cloudflare so you need to connect using cfscraper. Someone has already provided the python code here https://github.com/n0mad01/node.bittrex.api/issues/67#issuecomment-343736593

**Comment by skyl** [comment]
@dparlevliet wow, thanks. Works perfectly. Where's the tip jar?

**Comment by dparlevliet** [comment]
Tips are welcomed, just buy any altcoin on the Bittrex market. You'll likely find me in one of them ;)

**Comment by blockzen** [comment]
Hey anybody know how to apply this fix to the node.js api?

**Comment by dparlevliet** [comment]
@blockzen it already exists there, check you have the right package and version

**Comment by RikardoPons** [comment]
This is the url
http://bittrex.com/Api/v2.0/pub/market/GetTicks?marketName=BTC-ZEC&tickInterval=hour

It was working fine but now I got the same issue 403 Forbidden sometimes I get captcha page. 
This is annoying. does Bittrex support get candles in his Public API?  Thanks

**Comment by blockzen** [comment]
Could anyone help me update my bittrex Api will pay in your choice of crypto.  Thanks.

**Comment by karimhossenbux** [comment]
@blockzen what code do you have right now to get candles?

**Comment by blockzen** [comment]
not sure what you mean?  I don't think im connecting to the server right I get this error in terminal:

{ unhandled_data: {} }

**Comment by karimhossenbux** [comment]
I meant show us some code to understand what you are trying to do. Not only the response you got from the api

**Comment by blockzen** [comment]
///////////////bittrex socket connect///////////////////
console.log("position")
const websocketsclient = bittrex.websockets.listen( function( data ) {
	console.log(data);
	if (data.M === 'updateSummaryState') {
		data.A.forEach(function(data_for) {
			data_for.Deltas.forEach(function(marketsDelta) {
				let pair = marketsDelta.MarketName;
				let last = parseFloat(marketsDelta.Last);
				let bid = parseFloat(marketsDelta.Bid);
				let ask = parseFloat(marketsDelta.Ask);
				let bittrexObj = {};
				bittrexObj['pair'] = pair;
				bittrexObj['price'] = last;
				bittrexObj['bid'] = bid;
				bittrexObj['ask'] = ask;
				io.emit('bittrex', bittrexObj);
			});
		});
	}
});

**Comment by blockzen** [comment]
via websockets

**Comment by karimhossenbux** [comment]
Could you try something like this:

```
bittrex.websockets.client(function (wsclient) {
  wsclient.serviceHandlers.connected = function (connection) {
    wsclient.call('CoreHub', 'SubscribeToSummaryDeltas').done(function (err, result) {
      if (err) {
        return console.error(err)
      }
      if (result === true) {
        console.log('Websocket subscribed for ticker')
        bittrex.websockets.listen(function (data, client) {
          if (data.M === 'updateSummaryState') {
            data.A.forEach(function (dataFor) {
              console.log(dataFor)
            })
          }
        })
      }
    })
  }
})
```
